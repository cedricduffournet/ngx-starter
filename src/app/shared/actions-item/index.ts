export * from './actions-item.module';
