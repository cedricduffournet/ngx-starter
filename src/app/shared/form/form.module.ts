import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';

import { TranslateModule } from '@ngx-translate/core';
import { TextInputComponent } from '@app/shared/form/text-input';

@NgModule({
  imports: [CommonModule, ReactiveFormsModule, TranslateModule],
  exports: [TextInputComponent],
  declarations: [TextInputComponent]
})
export class FormModule {}
