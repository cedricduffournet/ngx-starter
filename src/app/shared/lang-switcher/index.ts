export * from './lang-switcher.module';
