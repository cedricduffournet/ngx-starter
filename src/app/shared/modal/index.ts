export * from './modal-wrapper.module';
