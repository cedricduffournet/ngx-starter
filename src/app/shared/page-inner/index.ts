export * from './page-inner.module';
