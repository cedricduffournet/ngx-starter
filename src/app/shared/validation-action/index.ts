export * from './validation-action.module';
