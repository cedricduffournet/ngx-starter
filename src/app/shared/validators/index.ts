export * from './areEqual.validator';
export * from './password.validator';
export * from './email.validator';
