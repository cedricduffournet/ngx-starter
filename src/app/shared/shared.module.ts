import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { TranslateModule } from '@ngx-translate/core';

import { ActionsItemsModule } from '@app/shared/actions-item';
import { PageInnerModule } from '@app/shared/page-inner';
import { IconsModule } from '@app/shared/icons';
import { FormModule } from '@app/shared/form';
import { ValidationActionModule } from '@app/shared/validation-action';
import { ModalWrapperModule } from '@app/shared/modal';
import { ButtonModule } from '@app/shared/button';

const MODULES_EXPORT = [
  CommonModule,
  ReactiveFormsModule,
  TranslateModule,
  ActionsItemsModule,
  PageInnerModule,
  IconsModule,
  FormModule,
  ValidationActionModule,
  ModalWrapperModule,
  ButtonModule
];

@NgModule({
  exports: MODULES_EXPORT
})
export class SharedModule {}
