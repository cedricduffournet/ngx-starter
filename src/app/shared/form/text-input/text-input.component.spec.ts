import { Component } from '@angular/core';
import { ReactiveFormsModule, FormControl, Validators } from '@angular/forms';
import {
  ComponentFixture,
  TestBed,
  fakeAsync,
  async,
  tick
} from '@angular/core/testing';
import { TranslateModule } from '@ngx-translate/core';
import { FormModule } from '@app/shared/form';
import { validateEmail, validatePassword } from '@app/shared/validators';

@Component({
  template: `
    <app-text-input
      [helpText]="helpText"
      [label]="label"
      [showError]="showError"
      [placeholder]="placeholder"
      [errorText]="errorText"
      [formGroupClass]="formGroupClass"
      [type]="type"
      [formControl]="controlTest"
      [submitted]="submitted"
    ></app-text-input>
  `
})
export class TextInputTestComponent {
  label = '';
  helpText = '';
  showError = false;
  placeholder = '';
  errorText = '';
  formGroupClass = '';
  submitted = false;
  type = 'text';
  controlTest: FormControl = new FormControl();
}

describe('TextInputComponent', () => {
  let fixture: ComponentFixture<TextInputTestComponent>;
  let component: TextInputTestComponent;
  let inputElement: any;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [FormModule, ReactiveFormsModule, TranslateModule.forRoot()],
      declarations: [TextInputTestComponent]
    });

    fixture = TestBed.createComponent(TextInputTestComponent);
    component = fixture.componentInstance;
    inputElement = fixture.nativeElement.querySelector('input');
  });

  it('should be created', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should init form controle with text', () => {
    component.controlTest = new FormControl('this is my text');

    fixture.detectChanges();

    expect(inputElement.value).toEqual('this is my text');
  });

  it('should get new value when subscribe to valueChanges', async(() => {
    component.controlTest = new FormControl();

    fixture.detectChanges();
    component.controlTest.valueChanges.subscribe((value: string) => {
      expect(value).toBe('new value');
    });

    inputElement.value = 'new value';
    inputElement.dispatchEvent(new Event('input'));
  }));

  it('input should be disabled', () => {
    component.controlTest = new FormControl({ value: '', disabled: true });

    fixture.detectChanges();

    expect(inputElement.disabled).toBeTruthy();
  });

  it('input should display required error message', () => {
    component.controlTest = new FormControl('', Validators.required);
    component.submitted = true;

    fixture.detectChanges();
    const error = fixture.nativeElement.querySelector('.error-required');
    expect(error).toBeTruthy();
  });

  it('input should display email error message', () => {
    component.controlTest = new FormControl('error_mail', validateEmail);
    component.submitted = true;

    fixture.detectChanges();
    const error = fixture.nativeElement.querySelector('.error-email');
    expect(error).toBeTruthy();
  });

  it('input should display error password', () => {
    component.controlTest = new FormControl('1234', validatePassword);
    component.submitted = true;

    fixture.detectChanges();
    const error = fixture.nativeElement.querySelector('.error-password');
    expect(error).toBeTruthy();
  });

  it('input should display error max length', () => {
    component.controlTest = new FormControl(
      'text is longer than 10 characters',
      Validators.maxLength(10)
    );
    component.submitted = true;

    fixture.detectChanges();
    const error = fixture.nativeElement.querySelector('.error-max-length');
    expect(error).toBeTruthy();
  });

  it('input should display custom error', () => {
    component.showError = true;
    component.errorText = 'Error';

    fixture.detectChanges();
    const error = fixture.nativeElement.querySelector('.custom-error');
    expect(error).toBeTruthy();
  });

  it('input should be type password', () => {
    component.type = 'password';
    fixture.detectChanges();
    expect(inputElement.type).toBe('password');
  });

  it('input should should have placeholder', () => {
    component.placeholder = 'placeholder';
    fixture.detectChanges();
    expect(inputElement.placeholder).toBe('placeholder');
  });
});
