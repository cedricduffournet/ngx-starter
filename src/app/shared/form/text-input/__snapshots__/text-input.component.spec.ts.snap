// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInputComponent should be created 1`] = `
<ng-component
  controlTest={[Function FormControl]}
  errorText=""
  formGroupClass=""
  helpText=""
  label=""
  placeholder=""
  showError="false"
  submitted="false"
  type={[Function String]}
>
  <app-text-input
    class="ng-untouched ng-pristine ng-valid"
    ng-reflect-error-text=""
    ng-reflect-form="[object Object]"
    ng-reflect-form-group-class=""
    ng-reflect-help-text=""
    ng-reflect-label=""
    ng-reflect-placeholder=""
    ng-reflect-show-error="false"
    ng-reflect-submitted="false"
    ng-reflect-type="text"
  >
    <div
      class="form-group "
    >
      <label
        class="control-label"
      >
        
      </label>
      <input
        class="form-control"
        placeholder=""
        type="text"
      />
      
      
      
      
      
      
    </div>
  </app-text-input>
</ng-component>
`;
